hop/publish.py

```diff
@@ -38,18 +48,19 @@ def _main(args=None):
 
     stream = Stream(format="json", config=config)
     with stream.open(args.url, "w") as s:
-        for gcn in args.gcn:
-            extension = os.path.splitext(gcn)[1]
-
-            # parse based on GCN type
-            if extension == ".gcn3":
-                with open(gcn, "r") as f:
-                    gcn = GCNCircular.from_email(f)
-            elif extension == ".xml":
-                with open(gcn, "rb") as f:
-                    gcn = VOEvent.from_xml(f)
-            else:
-                raise ValueError(f"File extension {extension} not recognized")
-
-            # publish GCN
-            s.write(gcn.asdict())
+
+        model_loader = {"circular": GCNCircular.from_email_file,
+                        "voevent": VOEvent.from_xml_file,
+                        "blob": MessageBlob.from_text,
+                        }
```

Comment:
I think it may be cleaner if we move this outside of the stream context. Also, I think it may be nicer if it's formatted as such:
```
model_loader = {
    "circular": GCNCircular.from_email_file,
    "voevent": VOEvent.from_xml_file,
    "blob": MessageBlob.from_text,
}
```

PEP8 is a little more loose on this requirement but there's already precedent for this formatting in the argparse arguments and I'd like to be consistent.